lib/firmware-base/system-status.c

```diff
@@ -0,0 +1,94 @@
+#include <driver/timer.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/timers.h>
+
+#include <stdio.h>
+
+#define TIMER_DIVIDER 80  // Divide Base by 80
+
+
+// Timer group 0, timer 0 configuration
+timer_config_t timer_config = {
+	.alarm_en    = TIMER_ALARM_DIS,	     // Disable alarm
+	.intr_type   = TIMER_INTR_DISABLE,   // No timer interrupts
+	.counter_dir = TIMER_COUNT_UP,	     // Count up
+	.auto_reload = TIMER_AUTORELOAD_EN,  // Auto-reload
+	.divider = TIMER_DIVIDER  // Set divider for 1ms (assumption that base
+				  // runs at 80Mhz)
+};
+
+// Initialization
+void init_timer0()
+{
+	timer_init(TIMER_GROUP_0, TIMER_0, &timer_config);
+	timer_set_counter_value(TIMER_GROUP_0, TIMER_0, 0);
+	timer_start(TIMER_GROUP_0, TIMER_0);
+}
+
+volatile uint64_t core0_idle_cycles = 0;
+volatile uint64_t core1_idle_cycles = 0;
+
+// When idle
+void vAppliocationIdleHook(void)
+{
+	// For ISR to handle (?)
+	BaseType_t xHigherPriorityTaskWoken = pdFALSE;
+
+	// Which core? (assumption 1 cycle == 1 tick)
```

Comment:
The FreeRTOS tick is triggered by some kind of SysTick timer (in our case I believe it's `TIMER0`.